### ISSUE TITLE
Looking for documentation on loss function for NER

### ISSUE THREAD

**Issue by chaolyang** [explosion/spaCy#3360]
I was reading the guide [here](https://spacy.io/usage/training#section-basics) but couldn't find any details on the loss function NER uses. I recall I see it mentioned that it's a log loss. Could someone explain how it is calculated and used in the optimization, and can it be changed to something else? Just trying to understand the optimization better. Perhaps also point me to the code where the loss calculation happens? Thanks in advance!

## Which page or section is this issue related to?

https://spacy.io/usage/training#section-basics

**Comment by honnibal** [comment]
The code for the gradient of the loss is here: https://github.com/explosion/spaCy/blob/0367f864fe90dfa1dcdd0bfaf8f06dbcd5e97e45/spacy/syntax/_parser_model.pyx#L153 . The loss function is currently hard-coded, so you'll have to build the code to experiment with modifications.

The answer is a bit more complicated than you'd probably expect, because the NER uses an imitation learning objective. I think I've explained this somewhere, I'll try to find the comment.

The quick answer is that we can end up with multiple actions that are equally okay. For instance, let's say the parser incorrectly predicts `B-PERSON`. In this state, `I-PERSON` and `L-PERSON` can be equally good continuations, if neither actions would add any new errors. The multi-label log loss takes this into account, and lets us be agnostic about which of these labels we prefer in these situations.

**Comment by honnibal** [comment]
This thread might be helpful: https://github.com/explosion/spaCy/issues/881

I'm not sure how we can put this in the docs, as it's very technical and requires a lot of background.
